src/nbl/ui/CClipboardManagerXCB.cpp

```diff
@@ -0,0 +1,147 @@
+
+#include "nbl/ui/CClipboardManagerXCB.h"
+#include "nbl/ui/XCBConnection.h"
+
+#include <mutex>
+#include <string>
+#include <vector>
+
+#include <xcb/xproto.h>
+
+namespace nbl::ui
+{
+    std::string CClipboardManagerXCB::getClipboardText() {
+        {
+            std::unique_lock<std::mutex> lk(m_clipboardMutex);
+            m_clipboardResponseCV.wait_until(lk, std::chrono::system_clock::now() + std::chrono::seconds(1));
+        }
+        std::lock_guard lk(m_clipboardMutex);
+        std::string response = std::move(m_clipboardResponse);
+        m_clipboardResponse = std::string();
+        return response;
+    }
+
+
+    bool CClipboardManagerXCB::setClipboardText(const std::string_view& data) {
+        std::lock_guard lk(m_clipboardMutex);
+        m_savedClipboard = data;
+        return true;
+    }
+
+    void CClipboardManagerXCB::process(const IWindowXCB* window, xcb_generic_event_t* event) {
+        const auto& xcb = m_connection->getXcbFunctionTable();
+        
+        auto TARGETS = m_connection->resolveAtom(m_TARGETS);
+
+        switch(event->response_type & ~0x80) {
+            // XCB_ATOM
+            // Somone is requesting the clipboard data
+            case XCB_SELECTION_REQUEST: {
+                auto* sne = reinterpret_cast<xcb_selection_request_event_t*>(event);
```

Comment:
seems like this is a Window event, ergo `CWindowXCB` should probably inherit from `IClipboardManager` and you should have no `ClipboardManagerXCB`

Reply:
why didn't I do that originally ...